MediaBrowser.Server.Mono/Program.cs

```diff
@@ -83,16 +85,43 @@ private static void SetSqliteProvider()
             SQLitePCL.Batteries_V2.Init();
         }
 
-        private static ServerApplicationPaths CreateApplicationPaths(string applicationPath, string programDataPath)
+        private static ServerApplicationPaths CreateApplicationPaths(
+            string applicationPath,
+            string programDataPath)
         {
-            if (string.IsNullOrEmpty(programDataPath))
+            var appFolderPath = Path.GetDirectoryName(applicationPath);
+
+            if (!string.IsNullOrEmpty(programDataPath))
             {
-                programDataPath = ApplicationPathHelper.GetProgramDataPath(applicationPath);
+                return new ServerApplicationPaths(
+                    programDataPath,
+                    appFolderPath,
+                    Path.GetDirectoryName(applicationPath));
             }
 
-            var appFolderPath = Path.GetDirectoryName(applicationPath);
-
-            return new ServerApplicationPaths(programDataPath, appFolderPath, Path.GetDirectoryName(applicationPath));
+            var useDebugPaths = false;
+#if DEBUG
+            useDebugPaths = true;
+#endif
+            
+            var debugProgramDataPath = ConfigurationManager.AppSettings["DebugProgramDataPath"];
+            var releaseProgramDataPath = ConfigurationManager.AppSettings["ReleaseProgramDataPath"];
+            var operatingSystemInfoLookup = new OperatingSystemInformationLookup();
+            
+            var applicationPathHelper =
+                new ApplicationPathHelper(
+                    debugProgramDataPath,
+                    releaseProgramDataPath,
+                    operatingSystemInfoLookup,
+                    useDebugPaths);
+            
+            programDataPath = applicationPathHelper.GetProgramDataPath(applicationPath);
+            Directory.CreateDirectory(programDataPath);
```

Comment:
Moved `CreateDirectory(...)` outside of ApplicationPathHelper, as the `GetProgramDataPath` name doesn't reveal the fact that the directory is also created. I'm happy to either move this back into the method if we rename it to make it clear the directory is created, or leave it here.